jax/_src/api.py

```diff
@@ -788,7 +788,7 @@ def arg_spec(x):
       return aval, device
 
   @api_boundary
-  def lower(*args, **kwargs) -> stages.Lowered:
+  def lower(*args, _experimental_lowering_platform: Optional[str] = None, **kwargs) -> stages.Lowered:
```

Comment:
Just to note: we could make this an option to `jit` (maybe `_experimental_aot_lowering_platform`) to avoid kwarg collision here. We don't have to worry so long as the name is so long and hidden, but we should keep that in mind if this approach graduates into something more official.

Reply:
I don't really like adding temp arguments to jit. Since it is temporary, we can keep it on lower and then if we decide on an approach, we can pull this up to `jit`.

Adding it to `jit` has more implications (even in terms of code and threading it through) that I feel this approach currently will satisfy George's needs and keep the implementation fairly clean. WDYT?

Reply:
Yep, this was a remark for our potential future selves.

Reply:
I also do not think it should be in `jit` because that is for lowering-compilation-execution. But it can be in an intermediate place: `jit().with_lowering_platform('tpu').lower(*args)`. We could do that if this ever "graduates", but perhaps it won't ever.